icx-cert/src/pprint.rs

```diff
@@ -0,0 +1,102 @@
+use anyhow::{anyhow, Context, Result};
+use ic_types::HashTree;
+use serde::{de::DeserializeOwned, Deserialize};
+use sha2::Digest;
+
+/// Structured contents of the IC-Certificate header.
+struct StructuredCertHeader<'a> {
+    certificate: &'a str,
+    tree: &'a str,
+}
+
+/// A fully parsed replica certificate.
+#[derive(Deserialize)]
+struct ReplicaCertificate {
```

Comment:
Ah yes. I wanted that structure to be maintained by the Agent (and most likely `ic-types` itself), but this is fine I guess.